src/components/Admin/ManageSensors/DeleteOldDataModal.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  CheckboxGroup,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  useDisclosure,
+  Text,
+  Divider,
+  Flex,
+} from '@chakra-ui/react';
+import React, {useState} from 'react';
+import {useTranslation} from 'react-i18next';
+import {FaTrash} from 'react-icons/fa';
+import {useAuth} from '../../../contexts/AuthContext';
+import {firestore} from '../../../firebase';
+
+/**
+ * Modal that allows users to mark old data for deletion
+ * Includes button that makes the delete data modal pop up
+ */
+function DeleteOldDataModal(): JSX.Element {
+  // --------------- State maintenance variables ------------------------
+  const {isOpen, onOpen, onClose} = useDisclosure();
+  const [
+    isDownloadedDisclosureChecked,
+    setDownloadedDisclosureChecked,
+  ] = useState(false);
+  const [isUploadedDisclosureChecked, setUploadedDisclosureChecked] = useState(
+    false
+  );
+  const [
+    isIrreversibleDisclosureChecked,
+    setIrreversibleDisclosureChecked,
+  ] = useState(false);
+
+  const {isAdmin} = useAuth();
+  const allDisclosuresChecked =
+    isDownloadedDisclosureChecked &&
+    isUploadedDisclosureChecked &&
+    isIrreversibleDisclosureChecked;
+  const {t} = useTranslation('administration');
+
+  /**
+   * Resets modal state values before closing the modal.
+   */
+  function handleClose(): void {
+    // Modal state
+    setDownloadedDisclosureChecked(false);
+    setUploadedDisclosureChecked(false);
+    setIrreversibleDisclosureChecked(false);
+    onClose();
+  }
+
+  /**
+   * Marks data older than a defined time as ready for deletion by a Cloud Function
+   */
+  async function markOldDataForDeletion(): Promise<void> {
+    if (allDisclosuresChecked && isAdmin) {
+      const sensorsList = await firestore.collection('sensors').get();
+      const deletionDocRef = firestore.collection('deletion').doc('todo');
+
+      // Get current mapping or set to empty
+      const mapping =
+        (await deletionDocRef.get())?.data()?.deletionMap ??
+        Object.create(null);
+
+      const offset = 7;
+      const deleteBeforeDate = new Date();
+      deleteBeforeDate.setDate(deleteBeforeDate.getDate() - offset);
+
+      for (const sensorDoc of sensorsList.docs) {
+        mapping[sensorDoc.id] = deleteBeforeDate;
+      }
+
+      deletionDocRef.update({deletionMap: mapping});
+    }
+  }
+
+  /**
+   * Handle a submission of the deletion modal
+   */
+  function handleSubmit(): void {
+    markOldDataForDeletion().then(handleClose);
+  }
+
+  return (
+    <Box marginY={2}>
```

Comment:
This margin is causing the button to be slightly out of line with the other buttons
```suggestion
    <Box>
```